### ISSUE TITLE
False positive DocblockTypeContradiction for Class with array not initialized

### ISSUE THREAD

**Issue by andrewprofile** [vimeo/psalm#2916]
I found similar case in #1338

Example: https://psalm.dev/r/86bbb4bff0
```php
<?php
final class Clazz {
    /**
     * @var array
     */
    private $x;

    public function func1 (): void {
      	if($this->x === null) {
	    $this->x = [];
        }
    }

    public function __construct () {
        $this->func1();
    }
}
```
Output: 
```bash
Psalm output (using commit 8a1067e): 

ERROR: DocblockTypeContradiction - 9:11 - array<array-key, mixed> does not contain null

ERROR: PropertyNotSetInConstructor - 6:12 - Property Clazz::$x is not defined in constructor of Clazz and in any methods called in the constructor
```

**Comment by muglug** [comment]
This is not a false-positive – you're using a hacky way to check for initialisation that will break with property types: https://3v4l.org/OKNYH

Psalm supports indirect initialisation here: https://psalm.dev/r/9934acb89f

**Comment by andrewprofile** [comment]
Thank you, in fact my logic did not make sense, I fixed my code.
